ctlcmd/cmdbase.py

```diff
@@ -771,7 +772,176 @@ def globcomp(text):
     globlist = [file + '/' if os.path.isdir(file) else file for file in globlist]
     return globlist
 
+ 
+class rootfilecmd(controlcmd):
+  """
+  @brief commands that control saving data to the root file
+  
+  @ingroup cli_design
+  
+  @details Commands for data which needs to be saved to a root file, this is an 
+  update to the old savefilecmd class which only wrote to a text file. All functions
+  that wish to have their default save location overridden should simply change the 
+  DEFAULT_SAVEFILE static variable.
+  
+  The data of timestamp (ms), detector id, gantry coordinates (mm), measured bias voltage (mv),
+  measured SiPM temp (C), and measured pulser board temp (C) will always be in the root file. 
+  Extra branches with more data will be added based on the commands run. 
+  
+  The save file string can also include placeholder strings to have the filename
+  automatically be modified according other input arguments and/or the state of
+  the system. This allows the same command to be used for different
+  configurations while still have the results be saved in distinct files.
+  Placeholders will be specified in the angle braced. For the full list of
+  special placeholders, as well as how the file parsing is handled, see the
+  detailed documentation in the `ctlcmd.cmdbase.savefilecmd.parse` method.
+  """
+  ##TODO: update this to make the file location customizable
+  DEFAULT_ROOTFILE = 'SAVEROOT_<TIMESTAMP>'
+  def __init__(self, cmd):
+    print("initialize rootfilecmd")
+    controlcmd.__init__(self, cmd)
+    
+  def add_args(self):
+    print("add_args in rootfilecmd")
+    group = self.parser.add_argument_group(
+        "root file saving options", """Options for changing the root file location.
+        For more details, see the official documentation.""")
+    group.add_argument('--saveroot',
+                       type=str,
+                       default=self.DEFAULT_ROOTFILE,
```

Comment:
If you want the default filenames to work, you will need to add a static DEFAULT_ROOTNAME to all classes that inherit the `rootfilecmd` class.